### ISSUE TITLE
Unable to connect to server

### ISSUE THREAD

**Issue by xxxserxxx** [mavit/slimpris2#5]
Trying to connect to a remote server (remote, but on the LAN), slimpris2 reports:

```
2020-12-12 14:05:57,501 slimpris2 ERROR: Could not connect to LMS: Remote end closed connection without response
```

It retries repeatedly, but never connects.  I am able to log into the LMS web interface from the same computer I'm trying to run slimpris2, and two different Android devices -- each running squeezer -- are able to connect to and control the LMS server (also over the WLAN).

Is there a way I can turn on more debugging and find out exactly what network error slimpris2 is encountering?

OS: Linux 5.9.13-arch1-1  
slimpris2: github master, 6a74199256501, Tue Sep 29 09:46:01 2020  
LMS: 8.0.0, on Ubuntu Ulyana, over WLAN  
Python: 3.9.0

**Comment by mavit** [comment]
You could try `--debug`, but I'm not sure it'll be very helpful in this case.  You could try looking at the logs on LMS to see if it says anything.  A network sniffer like Wireshark is always an option.

It's worth double checking you've got the right IP address for the server in the config file.

Are you able to telnet to ports 9000 and 9090 on the LMS server from the computer you're running slimpris2 on?

**Comment by xxxserxxx** [comment]
I enabled DEBUG jsonrpc logging, and the request slimpris2 is sending is:

```
[20-12-13 08:16:10.6072] Slim::Web::JSONRPC::handleURI (74) handleURI(Slim::Web::HTTP::ClientConn=GLOB(0x5593829f4420))
[20-12-13 08:16:10.6079] Slim::Web::JSONRPC::handleURI (140) POST data: [{"id": 1, "method": "slim.request", "params": ["9c:b6:d0:96:7c:ab", ["status", "0", null, "tags:"]]}]
[20-12-13 08:16:10.6092] Slim::Web::JSONRPC::handleURI (158) JSON parsed procedure: {
  id => 1,
  method => "slim.request",
  params => ["9c:b6:d0:96:7c:ab", ["status", 0, undef, "tags:"]],
}
[20-12-13 08:16:10.6105] Slim::Web::JSONRPC::requestMethod (376) requestMethod(["9c:b6:d0:96:7c:ab", ["status", 0, undef, "tags:"]])
[20-12-13 08:16:10.6116] Slim::Web::JSONRPC::requestMethod (485) 9c:b6:d0:96:7c:ab: request not dispatchable!
[20-12-13 08:16:10.6122] Slim::Web::JSONRPC::handleClose (55) Closing any subscriptions for Slim::Web::HTTP::ClientConn=GLOB(0x5593829f4420)
```

The null -> undef in the request params is concerning, but I honestly don't know enough to be able to tell.  I can connect from the slimpris2 client computer to both 9000 and 9090.  Other clients are sending json payloads like this:

```
[{"id":1,"method":"slim.request","params":["00:01:c0:1f:af:87",["status","-",1,"tags:uB"]]}]
```

so what stands out to me is that a valid (or, accepted & processed) request has a `[string,string,int,string]` parameters, and the one from slimpris2 is `[string,int,nil,string]`.

Does that provide any indication of what's going wrong?

**Comment by mavit** [comment]
What's `9c:b6:d0:96:7c:ab`?  It should be the MAC address of the player that you want to control, but in the working request you have `00:01:c0:1f:af:87`.

**Comment by xxxserxxx** [comment]
I have multiple clients connecting to this server; Squeezer connects over jsonrpc, I believe, and there are two Android phones in the house running that.  `00:01:c0:1f:af:87` is one of the clients submitting a *good* request to the server that it is handling properly.  `9c:b6:d0:96:7c:ab` is slimpris2.  I included `00:01:c0:1f:af:87` as an example of the difference between an accepted and processed request and the one from slimpris2.

**Comment by mavit** [comment]
> `00:01:c0:1f:af:87` is one of the clients submitting a good request to the server that it is handling properly.

Are you sure?  Does this device happen to also be running a player?  But anyway, let's not worry about that, because...

> `9c:b6:d0:96:7c:ab` is slimpris2.

You need to specify the MAC address (or the name) of the player that you want to control, not the client's MAC address.

**Comment by xxxserxxx** [comment]
> **You need to specify the MAC address (or the name) of the player that you want to control, not the client's MAC address.**

Aha. Ok, that makes sense.  I had this in `~/.config/slimpris2/slimpris2.config`.  

```
10216» grep mac ~/.config/slimpris2/slimpris2.conf
mac = 9c:b6:d0:96:7c:ab
```

I changed it to the MAC of the player that I want to control, and now it gives me a different error, so I'll close this and open another one.

I added emphasis above in case anybody comes across this, the documentation being rather sparse (although, I'm sure for people who dig into the guts of LMS enough, it would have been obvious).
